src/traceroute/android.cpp

```diff
@@ -78,50 +85,60 @@ AndroidProber::AndroidProber(bool a, int port, event_base *c)
   sockfd_ = measurement_kit::socket_create(family, SOCK_DGRAM, 0);
   if (sockfd_ == -1) {
     cleanup();
-    throw std::runtime_error("Cannot create socket");
+    error_cb_(std::runtime_error("Cannot create socket"));
+    return;
   }
 
   if (setsockopt(sockfd_, level_sock, opt_recverr, &val, sizeof(val)) != 0 ||
       setsockopt(sockfd_, level_proto, opt_recvttl, &val, sizeof(val)) != 0 ||
       setsockopt(sockfd_, SOL_SOCKET, SO_REUSEADDR, &val, sizeof(val)) != 0) {
     cleanup();
-    throw std::runtime_error("Cannot set socket options");
+    error_cb_(std::runtime_error("Cannot set socket options"));
+    return;
   }
 
-  if (measurement_kit::storage_init(&ss, &sslen, family, NULL, port) != 0) {
+  if (measurement_kit::storage_init(&ss, &sslen, family, NULL, port_) != 0) {
     cleanup();
-    throw std::runtime_error("measurement_kit::storage_init() failed");
+    error_cb_(std::runtime_error("measurement_kit::storage_init() failed"));
+    return;
   }
   if (bind(sockfd_, (sockaddr *)&ss, sslen) != 0) {
     cleanup();
-    throw std::runtime_error("bind() failed");
+    error_cb_(std::runtime_error("bind() failed"));
+    return;
   }
 
   // Note: since here we use `this`, object cannot be copied/moved
   if ((evp_ = event_new(evbase_, sockfd_, EV_READ, event_callback, this)) ==
       NULL) {
     cleanup();
-    throw std::runtime_error("event_new() failed");
+    error_cb_(std::runtime_error("event_new() failed"));
```

Comment:
Could you please also add a return statement here for robustness? We may need to add code below that in the future, and it's more future-proof to have a return statement already in place.